src/bundle/Security/TwoFactor/Event/ThrottlingListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scheb\TwoFactorBundle\Security\TwoFactor\Event;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Exception\TooManyRequestsHttpException;
+use Symfony\Component\RateLimiter\RateLimiterFactory;
+
+final class ThrottlingListener implements EventSubscriberInterface
```

Comment:
Please remove the `final` keyword and add PHPDoc `@final` (check other classes for reference). That's how I do it in the library to keep myself the ability to mock things if necessary.